### PR TITLE
rename "azurerm_synapse_big_data_pool" to "azurerm_synapse_spark_pool"

### PR DESCRIPTION
**Pull request by njuCZ**
for PR https://github.com/terraform-providers/terraform-provider-azurerm/pull/7886

the service team suddenly come up with a requirement: rename it to "azurerm_synapse_spark_pool", they say bigdatapool is just api name, they hope the product name is "sparkpool"

### CODE REVIEW

File: azurerm/internal/services/synapse/synapse_spark_pool_resource.go [hunk 1]
```diff
@@ -206,55 +206,55 @@ func resourceArmSynapseBigDataPoolCreateUpdate(d *schema.ResourceData, meta inte
...
 	}
 
 	d.SetId(*resp.ID)
-	return resourceArmSynapseBigDataPoolRead(d, meta)
+	return resourceArmSynapseSparkPoolRead(d, meta)
 }
 
-func resourceArmSynapseBigDataPoolRead(d *schema.ResourceData, meta interface{}) error {
+func resourceArmSynapseSparkPoolRead(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*clients.Client).Synapse.BigDataPoolClient
```
Comment (WodansSon) [inline azurerm/internal/services/synapse/synapse_spark_pool_resource.go]: Can we change the client name to match the new resource name here?

```suggestion
	client := meta.(*clients.Client).Synapse.SparkPoolClient
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
a few minor comments but this otherwise LGTM 👍 

Note: this isn't a breaking change since this hasn't shipped yet

**Comment by njuCZ** [comment]
@tombuildsstuff thanks for your quick review. I have modified it, please have a check again.

**Review by tombuildsstuff — APPROVED** [review]
LGTM 👍

**Comment by tombuildsstuff** [comment]
Tests pass:

Screenshot 2020-08-20 at 12 45 57

**Pull request merged**